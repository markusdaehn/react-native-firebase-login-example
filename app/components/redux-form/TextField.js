import React, { Component } from 'react';
import { Field } from 'redux-form';
import Input from './Input';

export default class TextField extends Component {
  constructor(props) {
    super(props);
  }

  focus() {
    this.field.getRenderedcomponent().focus();
  }

  render() {
    const { name, placeholder, ...otherProps } = this.props;

    return (
      <Field
      name={name || 'text'}
      ref={(field) => this.field = field}
      withRef
      placeholder={placeholder || 'Enter text'}
      autoCorrect={false}
      component={Input}
      {...otherProps}
      />
    );
  }
}
