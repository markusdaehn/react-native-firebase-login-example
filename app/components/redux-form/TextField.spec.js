import 'react-native';
import React from 'react';
import TextField from './TextField';
import { shallow } from 'enzyme';

describe('TextField', function () {
  let text;
  describe('when rendered with no props', function () {
    beforeEach(function () {
      text = shallow(<TextField />)
    });

    it('should render correctly with default props', function () {
      expect(text).toMatchSnapshot();
    });
  });
  describe('when rendered with name and placeholder properties set', function () {
    beforeEach(function () {
      text = shallow(<TextField name='text-field-name' placeholder='text-field-placeholder'  />)
    });
    it('should render correctly with name and placeholder set to passed in props', function () {
      expect(text).toMatchSnapshot();
    });
  });
});
