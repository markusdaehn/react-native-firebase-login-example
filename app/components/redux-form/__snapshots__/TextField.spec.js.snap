// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField when rendered with name and placeholder properties set should render correctly with name and placeholder set to passed in props 1`] = `
<Field
  autoCorrect={false}
  component={[Function]}
  name="text-field-name"
  placeholder="text-field-placeholder"
  withRef={true}
/>
`;

exports[`TextField when rendered with no props should render correctly with default props 1`] = `
<Field
  autoCorrect={false}
  component={[Function]}
  name="text"
  placeholder="Enter text"
  withRef={true}
/>
`;
